File: dmoj/topyodeller.cpp
#include <bits/stdc++.h>
using namespace std;
#pragma GCC optimize "Ofast"
int main() {
    int N; int K;
    scanf("%i %i", &N, &K);

    int totalScores[N];
    int worstRanks[N];

    // Fill totalScores and worstRanks with 0
    for(int i = 0; i < N; i++)  {
        totalScores[i] = 0;
        worstRanks[i] = 0;
    }

    for(int rnd = 0; rnd < K; rnd++) {
        for(int plyr = 0; plyr < N; plyr++) {
            int scr;
            scanf("%i",  &scr);
            totalScores[plyr] += scr;
        }

        for(int j = 0; j < N; j++) {
            int r = 1;
            for(int l = 0; l < N; l++)
                if(totalScores[l] > totalScores[j]) r++;

            if(worstRanks[j] < r) worstRanks[j] = r;
        }
    }

    int highestScore = *max_element(totalScores, totalScores+N);

    for(int i = 0; i < N; i++)
        if(totalScores[i] == highestScore) 
            printf("Yodeller %i is the TopYodeller: score %i, worst rank %i \n", i+1, highestScore, worstRanks[i]);
}

File: dmoj/topyodeller.java
import java.util.Scanner;

import javax.lang.model.util.ElementScanner6;

public class topyodeller {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int K = sc.nextInt();

        int[] scores = new int[N];
        int[] worstRanks = new int[N];

        for(int i = 0; i < K; i++) {
            for(int j = 0; j < N; j++) {
                int score = sc.nextInt();
                scores[j] += score;
            }

            // Get rank
            for(int j = 0; j < N; j++) {
                int rank = 1;
                for(int c = 0; c < N; c++) {
                    if(scores[c] > scores[j]) rank++;
                }
                if(rank > worstRanks[j]) worstRanks[j] = rank;
            }

        }

        int bestScore = Integer.MIN_VALUE;

        for(int i = 0; i < N; i++) {
            if(scores[i] > bestScore) {
                bestScore = scores[i];
            }
        }
        for(int i = 0; i < N; i++) {
            if(scores[i] == bestScore) {
                System.out.println("Yodeller " + (i+1) +" is the TopYodeller: score " + bestScore + ", worst rank " +worstRanks[i]);
            }
        }
        
    }
}

